src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/models/RadPatch.java

```diff
@@ -1,45 +1,125 @@
 package network.radicle.jetbrains.radiclejetbrainsplugin.models;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import git4idea.repo.GitRepository;
 
 import java.time.Instant;
 import java.util.List;
 
 public class RadPatch {
     public GitRepository repo;
-    public String patchId;
-    public String author;
+    public String id;
     public String title;
+    public Author author;
     public String description;
-    public State state;
     public String target;
+    public List<String> tags;
+    public State state;
     public List<Revision> revisions;
 
-    public RadPatch(GitRepository repo, String patchId, String author, String title, String description, State state, String target,
-                    List<Revision> revisions) {
-        this.repo = repo;
-        this.patchId = patchId;
-        this.author = author;
+    public record Author(String id) {
+    }
+
+    public RadPatch(String id, String title, Author author, String description, String target, List<String> tags,
+                    State state, List<Revision> revisions) {
+        this.id = id;
         this.title = title;
+        this.author = author;
         this.description = description;
-        this.state = state;
         this.target = target;
+        this.tags = tags;
+        this.state = state;
         this.revisions = revisions;
     }
 
+    public RadPatch() {
+    }
+
     public enum State {
-        OPEN, CLOSED, MERGED
+        OPEN("Open"),
+        CLOSED("Closed"),
+        MERGED("Merged"),
+        ARCHIVED("Archived");
+
+        public final String status;
+
+        State(String status) {
+            this.status = status;
+        }
+
+        @JsonCreator
+        public static State forValues(@JsonProperty("status") String status) {
+            for (State state : State.values()) {
+                if (state.status.toLowerCase().equals(status)) {
+                    return state;
+                }
+            }
+            return null;
+        }
     }
 
-    public record Revision(String patchId, String description, String base, String commitHash, Instant created, List<String> refs,
-                           List<Comment> comments, List<Merge> merges) {
+    public record Revision(String id, String description, String base, String oid, List<String> refs,
+                           List<Merge> merges, Instant timestamp,
+                           List<Discussion> discussions, List<String> reviews) {
+
+        @Override
+        public String toString() {
```

Comment:
this override does not really offer much, it's only a simulation of json (that can easily fail, e.g. in cases of null).
It's better to use jackson straight away for json serialization, so please remove all overriden `toString` from records.